api/views/personal_info.py

```diff
@@ -1,9 +1,25 @@
-from rest_framework import viewsets
+from rest_framework import status, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.serializers import BaseSerializer
 
 from api.models import PersonalInfo
-from api.serializers import PersonalInfoSerializer
+from api.serializers import PersonalInfoCheckExistenceSerializer, PersonalInfoSerializer
 
 
 class PersonalInfoViewSet(viewsets.ModelViewSet):
     queryset = PersonalInfo.objects.all()
-    serializer_class = PersonalInfoSerializer
+
+    def get_serializer_class(self) -> type[BaseSerializer]:
+        if self.action == "check_existence":
+            return PersonalInfoCheckExistenceSerializer
+        return PersonalInfoSerializer
```

Comment:
That's interesting. so you're basically using one `ViewSet` for 2 different purposes. Well why not.

Reply:
You could read here about it - https://www.django-rest-framework.org/api-guide/viewsets/#marking-extra-actions-for-routing

It's kinda breaking "rest", but still good for additional routes inside one "api resource"